pkg/ottl/ottlfuncs/func_replace_pattern_test.go

```diff
@@ -45,15 +45,15 @@ func Test_replacePattern(t *testing.T) {
 		{
 			name:    "replace regex match (with hash function)",
 			target:  target,
-			pattern: `passwd\=[^\s]*(\s?)`,
+			pattern: `passwd\=([^\s]*)(\s?)`,
```

Comment:
Can you add more tests here and for `replace_all_patterns` for:
1. Hash function with no `$1` in the replace value.
2. Hash function where `$1` is used but the regex does not contain any capture groups
3. No hash function where `$1` is used but the regex does not contain any capture groups

Reply:
Added tests for all 3 scenarios.

Reply:
Please do the same for `replace_all_patterns`

Reply:
added tests for `replace_all_patterns`